framework/Source/Operations/CBAddBlend.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  CBAddBlend.swift
+//
+//
+//  Created by Chiaote Ni on 2024/6/14.
+//
+
+import Foundation
+
+// It's the same as the AddBlend in GPUImage.
+// To create a separate class because the original design of GPUImage is not for executing under the async/await environment, and will easily cause a deadlock.
+// For more details, see here: https://www.notion.so/piccollage/await-newTexture-3962c12cc694452faeaa8210760898f4
+// Therefore, to run the processImage with async, this class override the newTextureAvailable function to achieve this goal to avoid the deadlock.
+// Also, follow the implementation of LookupFilter, I create an blendImageInput for this filter, which allow us to use the same interface in PicCollage when we migrate from using GPUImage3 to using CBVisualEffectBuiltIn
+public class CBAddBlend: BasicOperation {
+
+    public var intensity: Float = 1 { didSet { uniformSettings["intensity"] = intensity } }
+
+    public var blendImageInput: PictureInput? {
+        didSet {
+            inputTextures.removeValue(forKey: 1)
+            blendImageInput?.addTarget(self, atTargetIndex: 1)
```

Comment:
question: should we do some clean up in `willSet` for old input?

eg. remove the target and source

Reply:
We don't have to remove the source by ourselves, but yeah, forcing the old source to remove its target is nice to have.
I'll add and modify the didSet to the following, thanks!
```Swift
didSet {
    inputTextures.removeValue(forKey: 1)
    lookupImage?.addTarget(self, atTargetIndex:1)
    oldValue?.removeAllTargets()
}
```

Reply:
Would it be better to do this in willSet instead of using oldValue?

Reply:
Do you mean this?
```Swift
willSet {
    inputTextures.removeValue(forKey: 1)
    blendImageInput?.removeAllTargets()
}
didSet {
    lookupImage?.addTarget(self, atTargetIndex:1)
}
```

Reply:
In the didSet
should be sometihg like this?
```swift
 blendImageInput?.addTarget(self, atTargetIndex: 1)
 ```

Reply:
Sorry, I didn't get it; what's the difference between yours and my last one @@??
https://github.com/cardinalblue/GPUImage3/pull/20#discussion_r1639576790

Reply:
You edited the comment.

It was in the didSet `lookImage?.addTarget(self, atTargetIndex:1)`

Reply:
ohoh, yes it was the name that be used in lookup filter 😅
So we're saying the same thing 👍
Sure, I can do this in the next commit